cmd/restic/cmd_backup.go

```diff
@@ -427,6 +429,18 @@ func runBackup(opts BackupOptions, gopts GlobalOptions, args []string) error {
 			return false
 		}
 
+		if opts.ExcludeCaches && fi.Name() != "CACHEDIR.TAG" {
+			d := filepath.Dir(item)
+			_, err := os.Stat(filepath.Join(d, "CACHEDIR.TAG"))
```

Comment:
This is not enough, the file also needs to start with the following string: 
```
Signature: 8a477f597d28d172789f06886806bc55
```

Details here: http://www.brynosaurus.com/cachedir/spec.html

I'd suggest adding a separate function like this:
```
func isCacheFile(filename string) bool {}
```

The function then tests if there is a valid `CACHEDIR.TAG` in the same directory (and the file is not the cachedir tag file itself). That way the logic is nicely abstracted away from the select function. What do you think?

If we think it is too expensive to read the file every time we could also add a global cache (protected by a mutex) of known cache dirs and non-cache dirs. But let's add this first and worry about performance when it becomes an issue.